docs/conventions/component.md

```diff
@@ -0,0 +1,527 @@
+
+```twig
+{#
+  Parameters:
+    - "body" (string) (default: '')
+    - "author" (string) (default: '')
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: ''): format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
+#}
+
+{# Internal properties #}
+
+{% set _body = body|default('') %}
+{% set _author = author|default('') %}
+{% set _css_class = 'ecl-blockquote' %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~'"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result  #}
+
+<blockquote class="{\{ _css_class }}"{\{ _extra_attributes|raw }}>
+  <p class="ecl-paragraph ecl-blockquote__body">{{ _body }}</p>
+  {% if _author is not empty %}
+  <footer class="ecl-blockquote__author">―&thinsp;<cite>{{ _author }}</cite></footer>
+  {% endif %}
+</blockquote>
+```
+
+### Systems component
+
+There are 2 possibilities for system-based components: they use the exact same markup as their corresponding generic/base component, or they alter it.
+
+In both cases, **Twig file should always start with available parameters**.
+
+#### Same markup
+
+This is the easy case: system component just has to include generic Twig file.
+
+Example:
+_(File name: ec-component-blockquote.twig)_
+
+```twig
+{#
+  Parameters:
+    - "body" (string) (default: '')
+    - "author" (string) (default: '')
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: ''): format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
+#}
+
+{% include '@ecl/generic-component-blockquote' %}
+```
+
+#### Different markup
+
+As soon as there is a difference in markup (even a small one), specific component has to provide the whole markup (instead of using generic one).
+This is the same structure as generic Twig file.
+
+As we lose the advantage of generic components here, it is recommended, when possible, to keep the exact same markup for different systems.
+
+## Javascript
+
+Not all components require a dedicated javascript file. Most of them have simple enough behavior, that could be handled with HTML/CSS only. The following rules only concern component with more complex (js) behavior.
+
+Apart from [rules that should be applied to all js files](javascript), there are some specifications for components:
+
+### Generic component
+
+Generic component should export a function to handle specific behavior.
+This function should have an explicit name, based on component name, and an action verb (generally "Init"). It should also use camel case.
+
+Example:
+_(File name: generic-component-navigation-menu.js)_
+
+```
+export const navigationMenuInit = ({ [...] } = {}) => {
```

Comment:
I would have used the verb first...

Thinking... we could have `ECL.navigationMenu()` (shortcut for `ECL.navigationMenu({ autoInit: true })`, with this `autoInit` parameter used in the revealing module pattern to automatically init the component. This allows us to manually init the module if needed: `ECL.navigationMenu({autoInit: false}).init()`

Reply:
Let's keep it simple for now.
It could be discussed in a follow-up ticket